tests/ai_pipeline/test_pipeline.py

```
@@ -0,0 +1,368 @@
+class TestSaveDiagnosticPlot:
+
+    @pytest.fixture()
+    def plot_inputs(self, dummy_bgr_image):
+        h, w = dummy_bgr_image.shape[:2]
+        crops = [(0, 0, w, h)]
+        mask = np.zeros((2, 512, 512), dtype=np.float32)
+        mask[0, :200, :200] = 0.9
+        mask[1, :100, :100] = 0.8
+        return dummy_bgr_image, crops, [mask], 0.707, None, 0.0
+
+    def _make_mock_plt(self):
```

Ruff (0.15.12)

[warning] Missing return type annotation for private function `_make_mock_plt`

(ANN202)